doc_src/cmds/set.rst

```diff
@@ -97,6 +97,7 @@ The scoping rules when creating or updating a variable are:
 
 - If a variable's scope is not explicitly set and there is no existing variable by that name, the variable will be local to the currently executing function. Note that this is different from using the ``-l`` or ``--local`` flag, in which case the variable will be local to the most-inner currently executing block, while without them the variable will be local to the function as a whole. If no function is executing, the variable will be set in the global scope.
 
+- Similarly, a variable explicitly set with ``-f`` or ``--function`` outside of any function (a top-level declaration or at the interactive prompt) will be assigned to the global scope instead.
```

Comment:
This is wrong. Function-scope outside of a function is simply the top-level local scope.

```fish
set -f foo bar

begin
    set -f foo bar2
end

set -S foo
```

will print:

```
$foo: set in local scope, unexported, with 1 elements
$foo[1]: |bar2|
```

Reply:
Hmm, I'm not sure what I did wrong but I tested this yesterday and *thought* I had confirmed the behavior before I even wrote the doc change. Thanks for the correction.